controllers/pravegacluster_controller.go

```diff
@@ -425,13 +425,24 @@ func (r *PravegaClusterReconciler) reconcileSegmentStoreService(p *pravegav1beta
 			}
 		}
 	} else {
-
+		updateService := false
 		if currentService.Spec.Ports[0].Port != headlessService.Spec.Ports[0].Port {
 			currentService.Spec.Ports[0].Port = headlessService.Spec.Ports[0].Port
 			currentService.Spec.Ports[0].TargetPort = headlessService.Spec.Ports[0].TargetPort
+			updateService = true
+		}
+		if len(currentService.Spec.Ports) == 1 {
+			currentService.Spec.Ports = append(currentService.Spec.Ports, headlessService.Spec.Ports[1])
+			updateService = true
+		} else if currentService.Spec.Ports[1].Port != headlessService.Spec.Ports[1].Port {
+			currentService.Spec.Ports[1].Port = headlessService.Spec.Ports[1].Port
+			currentService.Spec.Ports[1].TargetPort = headlessService.Spec.Ports[1].TargetPort
+			updateService = true
+		}
+		if updateService {
 			err = r.Client.Update(context.TODO(), currentService)
 			if err != nil {
-				return fmt.Errorf("failed to update headless service port (%s): %v", currentService.Name, err)
+				return fmt.Errorf("failed to update headless service admin port (%s): %v", currentService.Name, err)
```

Comment:
`failed to update headless service admin port` -> `failed to update headless service ports` maybe?